aggregator/capture/aggregator.go

```diff
@@ -49,43 +50,40 @@ func NewAggregator() Aggregator {
 
 func (agg *aggregator) Open() error { return nil }
 
-func (agg *aggregator) AddMetricWithPoliciesList(
+func (agg *aggregator) AddUntimed(
 	mu unaggregated.MetricUnion,
-	pl policy.PoliciesList,
+	sm metadata.StagedMetadatas,
 ) error {
-	// Clone the metric and policies to ensure it cannot be mutated externally.
+	// Clone the metric and metadatas to ensure it cannot be mutated externally.
 	mu = cloneMetric(mu)
-	pl = clonePoliciesList(pl)
+	sm = cloneStagedMetadatas(sm)
```

Comment:
Do we always need to clone this? If so could we just return new object from iterator? Or maybe the caller of AddUntimed should do the clone before sending the objects in?

Reply:
Yea the iterator might reuse staged metadatas to reduce allocation overhead (otherwise we might need to allocate a lot more for the common case), hence the need for cloning. The "capturing" aggregator is only for testing purposes so it's okay to always clone. The real aggregator only clones when the metadata changes which is a lot more efficient.